cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/AbstractMetricsProcessorService.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metrics.process;
+
+import co.cask.cdap.api.metrics.MetricsContext;
+import co.cask.cdap.metrics.store.MetricDatasetFactory;
+import com.google.common.util.concurrent.AbstractExecutionThreadService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Process metrics by consuming metrics being published.
+ */
+public abstract class AbstractMetricsProcessorService extends AbstractExecutionThreadService {
```

Comment:
It's better not to have this abstract base class. This class is doing very little and we know that the Kafka sub-class of this is going to be gone.